src/core/mesh/qgsmeshlayer.h

```diff
@@ -183,6 +183,32 @@ class CORE_EXPORT QgsMeshLayer : public QgsMapLayer
     //! Returns active vector dataset
     QgsMeshDatasetIndex activeVectorDataset() const { return mActiveVectorDataset; }
 
+    /**
+      * Interpolates the value on the given point from given dataset.
+      *
+      * Returns NaN for NaN values, for values outside range or when
+      * triangular mesh is not yet initialized on given point
+      *
+      * \since QGIS 3.4
+      */
+    QgsMeshDatasetValue datasetValue( const QgsMeshDatasetIndex &index, const QgsPointXY &point ) const;
```

Comment:
The "point" argument is in map coordinates rather than layer coordinates, right? That's worth noting in documentation (the assumption is that we pass layer coordinates)

Also, worth mentioning that it uses cached previously used triangular mesh and so if the layer has not been rendered previously (e.g. when used in a script) it would return invalid value